### DISCUSSION TITLE
`cus-cn.tex` 里的代码使用疑问

### DISCUSSION THREAD

**Discussion by xkwxdyy** [Sophanatprime/cus#38] [General]
本讨论主要讨论一些手册里的代码使用疑问。

**Comment by xkwxdyy** [comment]
## 关于 `veta` 和数值的输入

https://github.com/Sophanatprime/cus/blob/aa23963ee748c79f5a481fbf491532c3dea03fbd/cus-cn.tex#L515-L525

里面的 `\veta` 是什么含义呢？

还有一个是对于这种数值 `1/7` 感觉 `\texttt` 不是很“语义”，而且 `a/b` 为什么不像 `1/7` 一样的处理呢？

**Reply by Sophanatprime** [reply]
`\veta` 用来标识一个变量名，就是 `\meta` 去掉两端的 angle。这里 `a` 和 `b` 表示变量，所以用 `\veta`。`1/7` 是 `\cusemojilowerratio` 存储的值，是一段代码，所以用 `\texttt`，更加“准确”的应当是 `\verb`。

**Reply by xkwxdyy** [reply]
`meta` 本身是有含义的，`veta` 的由来是单纯从 `meta` 来+`\V` 的效果？

**Reply by Sophanatprime** [reply]
`\veta` 只是去掉 `\meta` 两端的 angle，其它的完全一样，除了使用不同的格式命令（`\cus@doc@veta@format` 和 `\cus@doc@meta@format`）。

**Reply by xkwxdyy** [reply]
好的，了解了。请问一下什么时候该用 `\meta` 和 `\veta` 这种，什么时候该用 `\verb`？按照上面的说法，好像是变量的名一般用前者，变量的值用后者？

比如：`修改了某些键值 foo=bar 的 \verb|foo| 或者 \verb|bar|` 这里的后面应该用 `\verb`，但如果是抽象的  `foo = <content>`，那么后者就需要用 `\meta{content}` 了，如果要指里面的内容的话就要用 `\veta{content}`

这样想可以吗

**Reply by Sophanatprime** [reply]
在其它说明文档中，`\meta` 一般表示命令或键的参数，给这个参数起个别名，或者引用这个别名，倾向于表示某个参数。`\veta` 则是用于其它情形。不过一般用的是 `\Arg`，它和`\meta` 完全一样，语义上更清晰。

`\verb` 一般就是表示一段具体的代码。

**Reply by xkwxdyy** [reply]
好的，感谢，我还是尽量避免用根一点的 `\meta` ，多用一些语义清晰点的命令。

**Comment by xkwxdyy** [comment]
## 关于 `\UseList` 和 `\UseClist`

我注意到最近的 commit 更新了这两个命令
https://github.com/Sophanatprime/cus/blob/4bb96f344b1fa6f16b21cfa733f3a4482985073e/cusdoc.cls#L254-L271

感觉挺有用的，有没有考虑写到 `cus.sty` 里呢？

还有请问一下 list 和 clist 有什么区别吗，我只知道好像都是逗号分隔的列表

**Reply by Sophanatprime** [reply]
在本文档中，如果一类命令只有 `⟨list⟩` 的版本，那它就是逗号分隔的列表（`clist`），如果既有 `⟨list⟩` 又有 `⟨clist⟩`，那么前者是普通的 `token list`，后者才是逗号分隔的列表。不过这样指代不明，很容易造成误解，未来的版本应该写明是哪种。

比如 `\UseList{a b c}` 和 `\UseClist{a, b, c}` 一样，`\UseList{ {abc} {123} {ABC} }` 和 `\UseClist{ abc, 123, ABC }` 一样。

**Reply by xkwxdyy** [reply]
大致了解了，感谢。

**Comment by xkwxdyy** [comment]
## 关于语法糖 `<&...> `

手册里的解释是
> `<&...>` — 当文本环境中 `<` 紧跟 `&` 时，`...` 被视为可选值;

不太知道这是什么意思。

我试了一下
```latex
\begin{syntax}
  \verb|\foo| <\TF>
  \verb|\foo| <\TTF>
  \verb|\foo| <\TFF>
  \verb|\foo| <&\TF>
  \verb|\foo| <&\TTF>
  \verb|\foo| <&\TFF>
\end{syntax}
```
![image](https://github.com/Sophanatprime/cus/assets/87134063/46cfe5dd-7aeb-4854-8035-675688d01a85)

加了 `&` 的感觉就是改了字体？什么时候需要加 `&` 呢？不是很明白。

`\TF`的使用前面要加 `&` 吗？如果是的话，为什么不直接把 `\TF` 类型的cs 定义打个 hook 之类的呢？

**Reply by Sophanatprime** [reply]
如果键只接受某些固定的值就用 `&`，此时有确定的语意。

`\TF` 只是 `true\orbar false`，`\TTF` 只是 `\defaultval{true}\orbar false`，本身不带有可选值的语意，前面需要加 `&`。

> 为什么不直接把 `\TF` 类型的cs 定义打个 hook 之类的呢？

可选值不一定只有 `true` `false`，比如 `doc/cmd/hyper` 键就有 `true` `false` `raw` 三个，其它的键还可能不包含 `true` `false`。

**Reply by xkwxdyy** [reply]
> 比如 `doc/cmd/hyper` 键就有 `true` `false` `raw` 三个

那这种应该写 `<&\TF|raw>` 还是 `<&true|false|raw>` 呢

**Reply by Sophanatprime** [reply]
两个完全等价，前一个写起来更简单。

**Reply by xkwxdyy** [reply]
好的了解了，谢谢
